### PR TITLE
Add debug query to ProcessorError::DBStoreError

### PR DESCRIPTION
**Pull request by just-in-chang**
Adds an optional field for supplying a query for the `DBStoreError` to help with providing more context

### CODE REVIEW

File: rust/sdk-examples/src/utils/database.rs [hunk 1]
```diff
@@ -179,16 +179,19 @@ where
     debug!("Executing query: {:?}", debug_string);
     let conn = &mut pool.get().await.map_err(|e| {
         warn!("Error getting connection from pool: {:?}", e);
-        diesel::result::Error::DatabaseError(
-            diesel::result::DatabaseErrorKind::UnableToSendCommand,
-            Box::new(e.to_string()),
-        )
+        ProcessorError::DBStoreError {
```
Comment (larry-aptos) [inline rust/sdk-examples/src/utils/database.rs:182]: is it possible to move debug string construction only when erroring out? and is it possible to remove the clone?

Comment (banool) [reply rust/sdk-examples/src/utils/database.rs:182]: I think this only runs on error, this is a closure.

File: rust/sdk-examples/src/utils/database.rs [hunk 2]
```diff
@@ -179,16 +179,19 @@ where
...
+            query: Some(debug_string.clone()),
+        }
     })?;
     let res = final_query.execute(conn).await;
     if let Err(ref e) = res {
         warn!("Error running query: {:?}\n{:?}", e, debug_string);
     }
-    res
+    res.map_err(|e| ProcessorError::DBStoreError {
+        message: e.to_string(),
```
Comment (banool) [inline rust/sdk-examples/src/utils/database.rs:192]: You might get more context if you do this:
```suggestion
        message: format!("{:#}", e),
```

File: rust/sdk/src/utils/errors.rs [hunk 3]
```diff
@@ -9,5 +9,8 @@ pub enum ProcessorError {
     #[error("Poll Error: {message}")]
     PollError { message: String },
     #[error("DB Store Error: {message}")]
```
Comment (banool) [inline rust/sdk/src/utils/errors.rs:11]: You should also print the error here.

**Review by banool — APPROVED** [review]

File: rust/sdk-examples/src/utils/database.rs (hunk 1 above)
Comment (larry-aptos) [reply rust/sdk-examples/src/utils/database.rs:182]: oh my question is, can we move this into closure?  `let debug_string = diesel::debug_query::<Backend, _>(&final_query).to_string();`  my concern is the to_string creates a new copy of data, which is not only under error case.

Comment (just-in-chang) [reply rust/sdk-examples/src/utils/database.rs:181]: I just took out the debug logs and now it only prints on error 👍

Comment (larry-aptos) [reply rust/sdk-examples/src/utils/database.rs:181]: love this

**Pull request merged**